dali/kernels/tensor_shape_str.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_TENSOR_SHAPE_STR_H_
+#define DALI_KERNELS_TENSOR_SHAPE_STR_H_
+
+#include <dali/kernels/tensor_shape.h>
+#include <sstream>
+#include <string>
+
+namespace dali {
+namespace kernels {
+
+template <int ndim>
```

Comment:
Suggestion 1: put debug utilities inside a namespace (e.g. "debug") 
Suggestion 2: "str" -> "string_utils" or "debug_utils"

Reply:
1. I disagree. This is not just debug functionality and it doens't collide with anything. Stream operators in debug namespace are rather useless, because they'd force the user to write `using namespace` or import them explicitly or do something equally ugly.
2. Would `tensor_shape_print.h` be OK?